src/Streams/FileWriter.cpp

```diff
@@ -1,11 +1,32 @@
 #include "FileWriter.h"
 #include <stdexcept>
+#include <experimental/filesystem>
+namespace fs = std::experimental::filesystem;
 
 namespace Stream
 {
-	FileWriter::FileWriter(const std::string& filename) :
+	std::ios_base::openmode FileWriter::TranslateFlags(const std::string& filename, FileWriter::OpenMode openMode) {
+		if ((openMode & FileWriter::OpenMode::FailIfExist) != 0 && fs::exists(filename)) {
```

Comment:
Should there be a check here that raises and exception if both FailIfNoExist and FailIfExist are both specified together?

Reply:
I had considered that. An exception will be raised in that case, since the file must either exist or not exist, and so one of those two exception cases will be hit.

Unless of course a race condition allows the check to sneak past. I suppose I could do a single read of the file status, and use it with both flag checks. Or yes, I could test explicitly for both flags being set simultaneously and call the programmer an idiot. :wink: 

I'll look into cleaning it up a bit.

Reply:
Something like that???

For a console application consuming this library, I might add options to the user that line up with these flags when saving data. In that case, a nice error message would come in handy telling the end user they need to only use one of the flags. Also not too big a deal since they will probably figure it out on their own as you mentioned.